File: Data Structures & Algorithms/three-integer-sum/submission-4.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> res;
        sort(nums.begin(), nums.end());
        for(int i = 0; i < nums.size(); i++){
            if(i > 0 && nums[i]==nums[i-1]) continue;
            int tar = -nums[i];
            int l = i+1;
            int r = nums.size()-1;
            while(l < r){
                if(nums[l]+nums[r]==tar){
                   res.push_back(vector<int>{nums[i], nums[l], nums[r]});
                   r--;
                   l++;
                   while(l < r && nums[l] == nums[l-1]){
                    l++;
                   }
                   while(l < r && nums[r] == nums[r+1]){
                    r--;
                   }
                }
                else if(nums[l]+nums[r]<tar){
                    l++;
                }
                else{
                    r--;
                }
            }
        }
        return res;

    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        buc = [[] for i in range(len(nums)+1)]
        count = {}
        for n in nums:
            count[n]=1+count.get(n, 0)
        for ke,v in count.items():
            buc[v].append(ke)
        res = []
        for i in range(len(buc)-1, 0, -1):
            for j in buc[i]:
                res.append(j)
                if len(res) == k:
                    return res